### PR TITLE
Sessions Refactor

### PR DESCRIPTION
**Pull request by LHJE**
# Sessions
closes #95 

### `Sessions Controller`

Refactored to get the user from a SessonFacade

### `Sessions Facade`

Created with two methods, mirroring the work done on the GardenFacade

Spec made for these two methods.

### `Sessions Service`

Created to call and then parse the user json.

Tentative spec made for this one if we need it.

### Other changes

We updated all the files that had the Back End Api URL to have the Environment tag.

All Tests Passing.  (99.19%) coverage, with the usual suspects being the extra bits.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,10 +1,6 @@
 class SessionsController < ApplicationController
   def create
-    conn = Faraday.new(url: "https://solar-garden-be.herokuapp.com")
-      # refactor into a facade, environ variable    ^
-    response = conn.post('/api/v1/users', auth_hash)
-    json = JSON.parse(response.body, symbolize_names: true)
-    user = User.new(json[:data])
+    user = SessionFacade.session_details(auth_hash)
```
Comment (IamNorma) [inline app/controllers/sessions_controller.rb:3]: Nice cleanup here!

File: app/services/session_service.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class SessionService
+  def self.session_details(auth_hash)
+    get_parsed_json("/api/v1/users", auth_hash)
+  end
+
+  def self.get_parsed_json(url, auth_hash)
+    response = conn.post(url, auth_hash)
+    JSON.parse(response.body, symbolize_names: true)
```
Comment (IamNorma) [inline app/services/session_service.rb:8]: Really like how this was pulled into its own method

File: spec/services/session_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+describe SessionService do
```
Comment (IamNorma) [inline spec/services/session_service_spec.rb:3]: Is this test still being worked on?

Comment (LHJE) [reply spec/services/session_service_spec.rb:3]: No .... Because it's done!  The tests should be in there now!

File: app/poros/user.rb [hunk 4]
```diff
@@ -9,8 +9,8 @@ def initialize(data)
 
   def self.find(user_id)
     # call facade once facade + service pattens implemented
-    conn = Faraday.new(url: "https://solar-garden-be.herokuapp.com")
-    #this should be an environment variable  ^   
+    conn = Faraday.new(url: ENV['BE_URL'])
+    #this should be an environment variable  ^
```
Comment (leahriffell) [inline app/poros/user.rb:13]:
```suggestion
```

File: spec/facades/session_facade_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    expect(user).to be_a(User)
+  end
+
+  it "should generate session poros" do
+    data = {:data=>
+                    {
+                     :id=>"1",
+                     :type=>"garden",
+                     :attributes=>{
+                                   :email=> "big_tacos@tacos.com"
```
Comment (leahriffell) [inline spec/facades/session_facade_spec.rb:16]: a good kind of tacos

File: spec/facades/session_facade_spec.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+describe SessionFacade do
+  it "should return session details for a user" do
+    user = SessionFacade.session_details(stub_omniauth)
+
```
Comment (leahriffell) [inline spec/facades/session_facade_spec.rb:6]: If you wanted to test the whole response structure without having a huge list of assertions, you could add it to the spec>helper.rb file. Then you can just call that within any tests that need to reference the response for user call. There's some examples in there on garden and sensor if you're curious.

Comment (leahriffell) [reply spec/facades/session_facade_spec.rb:6]: JKKKK you already did that.

**Review by leahriffell — APPROVED** [review]
Just requested a commented out line deletion!

**Comment by LHJE** [comment]
Well dang it, it was working before some of those other PR's were pulled in :/

**Pull request merged**